plugin/evm/gossiper_orders.go

```diff
@@ -0,0 +1,182 @@
+package evm
+
+import (
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/subnet-evm/plugin/evm/message"
+	"github.com/ava-labs/subnet-evm/plugin/evm/orderbook"
+	"github.com/ava-labs/subnet-evm/plugin/evm/orderbook/hubbleutils"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+func (n *pushGossiper) GossipSignedOrders(orders []*hubbleutils.SignedOrder) error {
+	select {
+	case n.ordersToGossipChan <- orders:
+	case <-n.shutdownChan:
+	}
+	return nil
+}
+
+func (n *pushGossiper) awaitSignedOrderGossip() {
+	n.shutdownWg.Add(1)
+	go executeFuncAndRecoverPanic(func() {
+		var (
+			gossipTicker = time.NewTicker(ordersGossipInterval)
+		)
+		defer func() {
+			gossipTicker.Stop()
+			n.shutdownWg.Done()
+		}()
+
+		for {
+			select {
+			case <-gossipTicker.C:
+				if attempted, err := n.gossipSignedOrders(); err != nil {
+					log.Warn(
+						"failed to send signed orders",
+						"len(orders)", attempted,
+						"err", err,
+					)
+				}
+			case orders := <-n.ordersToGossipChan:
+				for _, order := range orders {
+					orderHash, err := order.Hash()
+					if err != nil {
+						log.Error("failed to hash order", "err", err)
+						continue
+					}
+					n.ordersToGossip[orderHash] = order
+				}
+				if attempted, err := n.gossipSignedOrders(); err != nil {
+					log.Warn(
+						"failed to send signed orders",
+						"len(orders)", attempted,
+						"err", err,
+					)
+				}
+			case <-n.shutdownChan:
+				return
+			}
+		}
+	}, "panic in awaitSignedOrderGossip", orderbook.AwaitSignedOrdersGossipPanicsCounter)
+
+}
+
+func (n *pushGossiper) gossipSignedOrders() (int, error) {
+	//
+	if (time.Since(n.lastOrdersGossiped) < minGossipOrdersBatchInterval) || len(n.ordersToGossip) == 0 {
+		return 0, nil
+	}
+	n.lastOrdersGossiped = time.Now()
+	orders := []*hubbleutils.SignedOrder{}
+	for orderHash, order := range n.ordersToGossip {
+		if len(orders) >= maxSignedOrdersGossipBatchSize {
+			break
+		}
+		orders = append(orders, order)
+		delete(n.ordersToGossip, orderHash)
+	}
+
+	now := time.Now().Unix()
+	selectedOrders := make([]*hubbleutils.SignedOrder, 0)
+	for _, order := range orders {
```

Comment:
2 loops seem unnecessary, can check for expiry in the first one itself?

Reply:
Yes, i'll check it in the first loop only

Reply:
done